### DISCUSSION TITLE
How do you get away with using SVGs still?

### DISCUSSION THREAD

**Discussion by NSExceptional** [alefragnani/vscode-bookmarks#407] [Q&A] [answered]
I'm writing my first VS Code extension and I noticed they forbid the use of SVGs (except those hosted by trusted domains).

Your extension provides its own SVGs. Doesn't `vsce` prevent you from publishing an extension with your own SVGs?

**Comment by alefragnani** [comment, marked as answer]
Hi @NSExceptional ,

Which `SVG` are you referring to? The one used on the **Side Bar** or the one on the **Gutter** (the Bookmark itself).

I’ve noticed a few dev complaints about SVG recently, but hadn’t any issues publishing my extensions. 

Maybe some `vsce` update / bug is the reason, not exactly your extension. I'm using v1.83.0, and the most recent is 1.85.1. Unfortunately there is no changelog available.

Hope this helps

**Reply by NSExceptional** [reply]
Odd, you shouldn't be able to with that version… vsce has apparently blocked SVGs as icons [as far back as 2017](https://github.com/microsoft/vscode-vsce/issues/183). More details [on this page.](https://code.visualstudio.com/api/working-with-extensions/publishing-extension)

Maybe @joaomoreno knows what's going on here?

**Reply by alefragnani** [reply]
That restriction was about SVG as extension's icon (which is displayed at the Marketplace) not the Side Bar.

The Side Bar can/must be a SVG, as described here https://code.visualstudio.com/api/extension-guides/tree-view?ref=hackernoon.com. I also remember it does follow a few rules (dimensions) but didn't find the right link

**Reply by NSExceptional** [reply]
Ah, interesting. I will open an issue on the relevant repo asking them to clarify the documentation then. Thank you!
